lagtraj/utils/gradient_calculation.py

```diff
@@ -15,35 +15,41 @@ def _boundary_gradients(x_array, y_array, val_array):
     len_lons = np.shape(val_array)[3]
     x_gradient_array = np.empty((len_temp, len_levels))
     y_gradient_array = np.empty((len_temp, len_levels))
-    dval_dx = np.empty((len_lats))
-    dval_dy = np.empty((len_lons))
     for this_time in range(len_temp):
         for this_level in range(len_levels):
-            # This calculates the x-gradient at each latitude
+            dxval_tot = 0.0
```

Comment:
I don't quite understand the need for changes in this file. Is that related to changes in how the mask is done? Or are these improvements for speed (reducing the number of calculations)?